### PR TITLE
Variable declaration, data types, numbers

### PR DESCRIPTION
**Pull request by IrinaMaslyukova**

### CODE REVIEW

File: Exercises/data-types-overview/data-types-overview.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+a = undefined;
+a = Symbol('Hi');
+a = {
+  1: 'First',
+  2: 'Second'
+};
+
+// 2.Use operator typeof on each of them.
+let a = typeof (123);
+a = typeof ('Hello');
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: you need to run it on variables that you should create above, not just reassigning `a`

File: Exercises/data-types-overview/data-types-overview.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+a = typeof (true);
+a = typeof (null);
+a = typeof (undefined);
+a = typeof (Symbol('Hi'));
+a = typeof ({
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: no need to assign to `a` result of typeof

File: Exercises/data-types-overview/data-types-overview.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  1: 'First',
+  2: 'Second'
+});
+
+//4. Demonstrate object's references.
+let obj = {
+  1: 'Apple',
+  2: 'banana',
+};
+const copyValue = {...obj};
```
Comment (lizarusi) [inline Exercises/data-types-overview/data-types-overview.js]: you are demonstrating how to workaround references :)

File: Exercises/numbers/numbers.js [hunk 4]
```diff
@@ -1,3 +1,37 @@
...
+{
+  const num = new Number(16);
+  console.log(num);
+}
+
+//2.Implement function extractNumbers which will leave only numbers in passed string.
+let num = '123rr';
+
+function extractNumbers(num) {
+  return parseInt(num);
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: mention that float numbers can appear

File: Exercises/numbers/numbers.js [hunk 5]
```diff
@@ -1,3 +1,37 @@
...
+}
+
+console.log(extractNumbers(num));
+
+//3.Declare a variable with decimal value and convert to binary, octal and hexadecimal.
+let num = new Number(10);
+let str_1 = num.toString(2);
+console.log(parseInt(str_1, 2));
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: no need to parse them again

File: Exercises/numbers/numbers.js [hunk 6]
```diff
@@ -1,3 +1,37 @@
...
+let str_2 = num.toString(8);
+console.log(parseInt(str_2, 8));
+let str_3 = num.toString(16);
+console.log(parseInt(str_2, 16));
+
+//4. use Math.random to generate random value from 0 to 100.
+let a = Math.round(Math.random());
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: you need to generate random value from 0 to 100. you are generating from 0 to 1

File: Exercises/numbers/numbers.js [hunk 7]
```diff
@@ -1,3 +1,37 @@
...
+let num = '123rr';
+
+function extractNumbers(num) {
+  return parseInt(num);
+}
+
+console.log(extractNumbers(num));
+
+//3.Declare a variable with decimal value and convert to binary, octal and hexadecimal.
+let num = new Number(10);
```
Comment (lizarusi) [inline Exercises/numbers/numbers.js]: always use `const` when variable not going to be change

**Review by lizarusi — CHANGES_REQUESTED** [review]

File: Exercises/data-types-overview/data-types-overview.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
+//1. Declare a variable on each type using two options: plain creation, creation with constructor.1
+let a = 123;
+let a_1 = new Number(123);
```
Comment (Mike-Savin) [inline Exercises/data-types-overview/data-types-overview.js:3]: Javascript variables should have names in camelCase, not in snake_case.

File: Exercises/data-types-overview/data-types-overview.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+let e = undefined;
+let f = Symbol('Hi');
+let g = {
+  1: 'First',
+  2: 'Second'
+};
+
+// 2.Use operator typeof on each of them.
+console.log(typeof (a), typeof (a_1), typeof (b), typeof (b_1), typeof (c),
+  typeof (c_1), typeof (d), typeof (e), typeof (f), typeof (g));
```
Comment (Mike-Savin) [inline Exercises/data-types-overview/data-types-overview.js:18]: You don't need brackets after typeof.

File: Exercises/numbers/numbers.js [hunk 10]
```diff
@@ -1,3 +1,36 @@
-function extractNumbers (text) {
-  return null;
-}
\ No newline at end of file
+//1.Assign by hand same value to 3 variables in binary, octal and hexadecimal format.
+{
+  const num = new Number(2);
+  console.log(num);
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js:4]: `new Number(x)` creates number with value x, not a value in x format.

File: Exercises/numbers/numbers.js [hunk 11]
```diff
@@ -1,3 +1,36 @@
...
+//2.Implement function extractNumbers which will leave only numbers in passed string.
+let number = '123jj';
+function extractNumbers(number) {
+  return parseFloat(number);
+}
+console.log(extractNumbers(number));
+
+//3.Declare a variable with decimal value and convert to binary, octal and hexadecimal.
+const num = new Number(10);
+let str_1 = num.toString(2);
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js:24]: names should be in camelCase

File: Exercises/numbers/numbers.js [hunk 12]
```diff
@@ -1,3 +1,36 @@
...
+console.log(str_1);
+let str_2 = num.toString(8);
+console.log(str_2);
+let str_3 = num.toString(16);
+console.log(str_3);
+
+//4. use Math.random to generate random value from 0 to 100.
+const min = 1
```
Comment (Mike-Savin) [inline Exercises/numbers/numbers.js:32]: Don't miss `;` at the end of the string

**Review by Mike-Savin — CHANGES_REQUESTED** [review]

**Review by Mike-Savin — CHANGES_REQUESTED** [review]
see my comments

**Pull request merged**